runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -7482,6 +7491,13 @@ class INTERPRETER_CLASS
 					goto done;
 				}
 #if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
+				/* NullRestricted field cannot be set to null.
+				 * The BytecodeInterpreter represents this as 0 for objects.
+				 */
+				if (J9_ARE_ALL_BITS_SET(flags, J9FieldFlagIsNullRestricted) && (0 == *_sp)) {
```

Comment:
Instead of using `(0 == *_sp),` can you use a local variable and compare it against NULL ? i.e. like the example a few lines above which checks if `objectref` is NULL.